File: backend/src/main/java/es/codeurjc/webapp14/dto/BasicProductSizeDTO.java
package es.codeurjc.webapp14.dto;

import java.util.List;

public record BasicProductSizeDTO(
    Long id,
    String name,
    String description,
    Double price,
    Integer stock,
    boolean outOfStock,
    int sold,
    String category,
    boolean imageBool,
    String imageUrl,
    List<SizeDTO> sizes
) {}

File: frontend/src/app/dtos/basicProductSize.dto.ts
import { SizeDTO } from "./size.dto";


export interface BasicProductSizeDTO {
	id?: number,
	name: string,
    description: string,
    price: number,
	stock: number,
    outOfStock: boolean,
    sold: number,
    category: string,
    imageBool: boolean,
    imageUrl: string,
    sizes: SizeDTO[]
}